ql/src/java/org/apache/hadoop/hive/ql/optimizer/lineage/Generator.java

```diff
@@ -59,11 +68,63 @@ public class Generator extends Transform {
 
   private static final Logger LOG = LoggerFactory.getLogger(Generator.class);
 
-  private final Set<String> hooks;
-  private static final String ATLAS_HOOK_CLASSNAME = "org.apache.atlas.hive.hook.HiveHook";
+  public static Generator fromConf(HiveConf conf) {
+    return new Generator(createFilterPredicateFromConf(conf));
+  }
+
+  private static final String ALL = "ALL";
+  private static final String NONE = "NONE";
+  private static final Map<HiveOperation, Function<ParseContext, Boolean>> filterMap;
 
-  public Generator(Set<String> hooks) {
-    this.hooks = hooks;
+  static {
+    Map<HiveOperation, Function<ParseContext, Boolean>> map = new HashMap<>();
+    map.put(HiveOperation.CREATETABLE, parseContext -> parseContext.getCreateTable() != null);
+    map.put(HiveOperation.CREATETABLE_AS_SELECT, parseContext -> parseContext.getQueryProperties().isCTAS());
+    map.put(HiveOperation.CREATEVIEW, parseContext -> parseContext.getQueryProperties().isView());
+    map.put(HiveOperation.CREATE_MATERIALIZED_VIEW,
+        parseContext -> parseContext.getQueryProperties().isMaterializedView());
+    map.put(HiveOperation.LOAD,
+        parseContext -> !(parseContext.getLoadTableWork() == null || parseContext.getLoadTableWork().isEmpty()));
+    map.put(HiveOperation.QUERY, parseContext -> parseContext.getQueryProperties().isQuery());
+    filterMap = Collections.unmodifiableMap(map);
+  }
+
+  static Predicate<ParseContext> createFilterPredicateFromConf(Configuration conf) {
+    Set<HiveOperation> operations = new HashSet<>();
+    boolean noneSpecified = false;
+    for (String valueText : conf.getStringCollection(HiveConf.ConfVars.HIVE_LINEAGE_STATEMENT_FILTER.varname)) {
+      if (ALL.equalsIgnoreCase(valueText)) {
+        return parseContext -> true;
+      }
+      if (NONE.equalsIgnoreCase(valueText)) {
+        noneSpecified = true;
+        continue;
+      }
+
+      HiveOperation enumValue = EnumUtils.getEnumIgnoreCase(HiveOperation.class, valueText);
+      if (enumValue == null) {
+        throw new EnumConstantNotPresentException(HiveOperation.class, valueText);
+      }
+      operations.add(enumValue);
+    }
+
+    if (noneSpecified) {
+      if (!operations.isEmpty()) {
+        throw new IllegalArgumentException("No other value can be specified when None is present!");
+      }
+      else {
+        return parseContext -> false;
+      }
+    }
+
+    return parseContext ->
+        operations.stream().anyMatch(hiveOperation -> filterMap.get(hiveOperation).apply(parseContext));
+  }
+
+  private final Predicate<ParseContext> statementFilter;
```

Comment:
Can you move that declaration and the constructor up so that the class can keep that pattern: fields -> constructors -> methods?

Reply:
Everything before `statementFilter` is static. Where should those static fields and methods go?

Reply:
I would really not add method declarations before field ones. 
What about static fields, instance fields, static initialization block, constructor, public methods, private methods?

Reply:
I usually move static members to the beginning of the file but I don't have string opinion about this.
I changed the order as you suggested.